app/celery/process_delivery_status_result_tasks.py

```diff
@@ -265,3 +276,266 @@ def sms_status_update(
+):
+    """Attempt retry sending notification.
+
+    Retry notification if within permissible limits and update notification.
+    If retry limit or retry window exceeded call sms_status_update with STATUS_REASON_UNDELIVERABLE
+
+    Args:
+        sms_status (SmsStatusRecord): The status record update
+        event_timestamp (str | None, optional): Timestamp the Pinpoint event came in. Defaults to None.
+        event_in_seconds (int, optional): How long since initial event. Defaults to 300
+
+    Raises:
+        NonRetryableException: Unable to update SMS retry count or update the notification
+    """
+
+    # avoid circular import
+    from app.notifications.process_notifications import send_notification_to_queue_delayed
+
+    notification = _get_notification(sms_status.reference, sms_status.provider, event_timestamp, event_in_seconds)
+
+    current_app.logger.info(
+        'Entering %s retryable failure logic to process event | reference: %s | notification_id: %s | current_status: %s | current_status_reason: %s | event_sms_status: %s | event_sms_status_reason: %s',
+        sms_status.provider,
+        sms_status.reference,
+        notification.id,
+        notification.status,
+        notification.status_reason,
+        sms_status.status,
+        sms_status.status_reason,
+    )
+
+    notification_retry_id = f'notification-carrier-sms-retry-count-{notification.id}'
+
+    try:
+        retry_count_redis_ttl = int(CARRIER_SMS_MAX_RETRY_WINDOW.total_seconds())
+        retry_count = update_sms_retry_count(notification_retry_id, ttl=retry_count_redis_ttl)
+    except Exception:
+        current_app.logger.error('Unable to retrieve value from Redis')
+        retry_count = None
+
+    if can_retry_sms_request(
+        notification.status, retry_count, CARRIER_SMS_MAX_RETRIES, notification.sent_at, CARRIER_SMS_MAX_RETRY_WINDOW
+    ):
+        retry_delay = get_sms_retry_delay(retry_count)
+
+        # need to roll notification status back to 'created' for requeue to work
+        # reference is also cleared to avoid race conditions while being requeued
+        try:
+            notification: Notification = dao_update_sms_notification_status_to_created_for_retry(
+                notification_id=notification.id,
+                notification_type=notification.notification_type,
+                cost_in_millicents=notification.cost_in_millicents + sms_status.price_millicents,
+                segments_count=sms_status.message_parts,
+            )
+
+            current_app.logger.info(
+                'Notification updated prior to requeue attempt | notification_id: %s | notification_status: %s | cost_in_milicents %s',
+                notification.id,
+                notification.status,
+                notification.cost_in_millicents,
+            )
+            statsd_client.incr(f'clients.sms.{sms_status.provider}.delivery.status.{sms_status.status}')
```

Comment:
Please move these statements to after the try/except unless you anticipate them raising exceptions.  For generally PEP8 compliance, "try" usually should only include statements that might raise exceptions.